x-pack/legacy/plugins/epm/server/datasources/create.ts

```diff
@@ -241,14 +172,73 @@ async function getConfig(pkgkey: string, dataset: Dataset): Promise<string> {
     isDatasetInput(entry, dataset.name)
   );
 
-  if (paths.length === 1) {
+  if (paths.length === 1 && Array.isArray(vars)) {
```

Comment:
Some packages would generates a `vars is not iterable` error, so this is a guard against that

Reply:
Reading this again, I don't know if/why this works. I'm going to remove it locally and see if I can still reproduce the "not iterable" error.

Reply:
This sounds like an invalid package to me. We should add a validation check in the registry to prevent this. Want to file an issue there?

Reply:
I added a note in 5dba6cc67743ed0962a3c27405e777371c9aae98. Vars is an optional param according to the registry struct. Marking it as optional in the TS type warns about this issue. I updated the TS type so EPM & the registry are in sync